ansible_base/authentication/social_auth.py

```diff
@@ -144,6 +153,24 @@ def ensure_strategy_in_args(self, args):
         return args
 
 
+class SocialAuthValidateCallbackMixin:
+    def validate(self, serializer, data):
+        # if we have an instance already and we didn't get a configuration parameter we are just updating other fields and can return
+        if serializer.instance and 'configuration' not in data:
+            return data
+
+        configuration = data['configuration']
+        if not configuration.get('CALLBACK_URL', None):
+            if not serializer.instance:
+                slug = generate_authenticator_slug(data['type'], data['name'])
+            else:
+                slug = serializer.instance.slug
+
+            configuration['CALLBACK_URL'] = reverse('social:complete', request=serializer.context['request'], kwargs={'backend': slug})
+
+        return data
+
+
 def create_user_claims_pipeline(*args, backend, **kwargs):
     from ansible_base.authentication.utils.claims import update_user_claims
 
```

Comment:
I see similar ...
```
aap_gw_1           | 2024-02-06 15:12:57,262 INFO     aap.gateway.authentication.util Loading Gateway social auth settings                                                                                                                                                                                                    
aap_gw_1           | 2024-02-06 15:12:57,271 INFO     ansible_base.authentication.utils.claims Creating mapping for user jctanner through authenticator github test                                                                                                                                                           
aap_gw_1           | 2024-02-06 15:12:57,271 DEBUG    ansible_base.authentication.utils.claims jctanner's groups: []                                                                                                                                                                                                          
aap_gw_1           | 2024-02-06 15:12:57,271 DEBUG    ansible_base.authentication.utils.claims jctanner's attrs: <bound method QuerySet.extra of <django.db.models.fields.related_descriptors.create_reverse_many_to_one_manager.<locals>.RelatedManager object at 0x7ffa2bfceb50>>                                           
aap_gw_1           | 2024-02-06 15:12:57,276 DEBUG    ansible_base.authentication.utils.claims access_allowed: True                                                                                                                                                                                                           
aap_gw_1           | 2024-02-06 15:12:57,276 DEBUG    ansible_base.authentication.utils.claims Setting new attribute access_allowed for jctanner                                                                                                                                                                              
aap_gw_1           | 2024-02-06 15:12:57,276 DEBUG    ansible_base.authentication.utils.claims claims: {'team_membership': {}, 'organization_membership': {}}                                                                                                                                                                 
aap_gw_1           | 2024-02-06 15:12:57,276 DEBUG    ansible_base.authentication.utils.claims Setting new attribute claims for jctanner                                                                                                                                                                                      
aap_gw_1           | 2024-02-06 15:12:57,276 DEBUG    ansible_base.authentication.utils.claims last_login_map_results: []                                                                                                                                                                                                     
aap_gw_1           | 2024-02-06 15:12:57,282 ERROR    ansible_base.authentication.utils.claims TODO: Fix reconciliation of user claims                                                                                                                                                                                        
aap_gw_1           | 2024-02-06 15:12:57,282 ERROR    ansible_base.authentication.utils.claims {'team_membership': {}, 'organization_membership': {}}                                                                                                                                                                         
aap_gw_1           | 2024-02-06 15:12:57,282 INFO     ansible_base.authentication.backend User jctanner logged in from github test  
```

Reply:
This can be fixed via ... 

```
# ansible_base/authentication/utils/claims.py
-    extra_data = user.authenticator_user.get(provider=database_authenticator).extra_data
-    results = create_claims(database_authenticator, user.username, extra_data, groups)
+    results = create_claims(database_authenticator, user.username, user.authenticator_user.extra, groups)
```

However, it breaks the ldap unit tests so further refactoring is necessary. John said to do that in a different PR.